### PR TITLE
Carry over extensions when expanding content references

### LINKED ISSUE #699: Content References should recursively inherit extensions
See https://chat.fhir.org/#narrow/stream/215610-shorthand/topic/Questionnaire.20item.20extensions/near/216143049
and https://chat.fhir.org/#narrow/stream/179252-IG-creation/topic/Clarification.20on.20contentReference/near/204963141

### PR DESCRIPTION
**Pull request by cmoesel**
This is one of those times that I over-complicated things, wrote a bunch of code that didn't work, then stepped back and found a simpler solution, which also didn't work, and then finally fixed that.  I have to admit, the regression checker also saved me from pushing up a solution that was majorly broken; so put this one in the win column for regression tests.

To manually test this, try this FSH (inspired by the Zulip thread):
```
Profile: AdvancedQuestionnaire
Parent: Questionnaire
* item.extension contains  minValue named minIntValue 0..1 MS

Instance: MyQuestionnaire
InstanceOf: AdvancedQuestionnaire
* status = #active
* item.linkId = "1"
* item.type = #group
* item.item.linkId = "1.1"
* item.item.type = #integer
* item.item.extension[minIntValue].valueInteger = 130
```

On the main branch, it will result in an error (path does not exist), but on this PR branch, it will succeed and the extension will be correctly generated in the JSON.

Fixes #699.

### CODE REVIEW

File: test/utils/PathUtils.test.ts
Comment (jafeltra) [inline test/utils/PathUtils.test.ts:1]: It looks like git got confused when merging these changes. I don't think we need this line since it's also included at the bottom of the import list. I'll let you decide which place you like it better though!

File: src/fhirtypes/ElementDefinition.ts
Comment (jafeltra) [inline src/fhirtypes/ElementDefinition.ts:1740]: I was trying to understand the case where we would filter out these recursive content reference paths, which led me to try taking out this line, which made me notice that it doesn't look like either test includes a check to ensure these are filtered out properly. Does it make sense to add that to one of the tests or add a new test? Or am is this too much of an edge case?

**Review by jafeltra — COMMENTED** [review]
This looks good to me! I tested it out with the example you provided and I ran the regression script on 58 known FSH projects and only one had meaningful changes, and they all were what I would expect. I just had one minor suggestion and a question below.

**Review by ngfreiter — COMMENTED** [review]
This looks good to me, all I noticed was what @jafeltra already said, so I will hold off on approving until those comments are addressed.

File: test/utils/PathUtils.test.ts
Comment (cmoesel) [reply test/utils/PathUtils.test.ts:1]: Hmmm... I kind of like it on line 1.  Plus that gives me more git cred.  But at the same time, @ngfreiter got his in first, so it would be kind of silly to have this PR do nothing in that file except move the import line so... I guess I'll let Nick's stay. :-(

File: src/fhirtypes/ElementDefinition.ts
Comment (cmoesel) [reply src/fhirtypes/ElementDefinition.ts:1740]: I'll try to cover it in a test.

**Comment by cmoesel** [comment]
OK, @jafeltra and @ngfreiter -- 36a5ab0 should address the comments.  @jafeltra -- you'll notice that if you uncomment that line, the test now fails.  I had a hard time wording the comment in the test, but hopefully it makes at least 51% sense?

**Review by ngfreiter — DISMISSED** [review]
This made my head hurt, but I think the test makes sense and looks good to me!

**Review by jafeltra — APPROVED** [review]
Agreed, my head hurts, but I think (?) this makes sense. Thanks for adding it.

**Comment by cmoesel** [comment]
I just reviewed the regression that @jafeltra found with `HL7NZ/cca#master`.  I don't think that's an intentional regression; I think something is wrong.  So... putting this back into _DRAFT_ until I figure it out.

**Comment by cmoesel** [comment]
OK @jafeltra and @ngfreiter -- this is ready for re-review.  Since your last review, I made the following changes:
* If a contains rule has multiple items, and there is an error on one of them (e.g., duplicate slice name), it will continue to process the others.  Before this it threw an exception that was caught to high up the chain and skipped all the other items.
* If a non-inline extension contains rules redeclares a slicename with the same extension URL, just treat it as a warning since it's basically a no-op.  It's possible the contains rule has a different cardinality, but the normal cardinality rules will still get applied, so if that ends up being a problem, an error will be produced anyway.
* Regression script now reports the number of errors returned by each SUSHI invocation so you can notice if the error count changes.  This also required a change to `app.ts` so the exit code is now the number of errors encountered (instead of always `1` if there are errors, no matter how many errors).

**Comment by cmoesel** [comment]
Oh, I forgot.  Also:
* If an extension is carried over from the source of a contentReference, it will now be reflected in the differential as well.  This is because the IG Publisher doesn't correctly produce the snapshot indicating carried over extensions unless we do so.

**Review by jafeltra — APPROVED** [review]
I'm still not 100% sure I understand everything that's happening here or what was originally going wrong with the HL7NZ/cca repo, but I think all the individual changes and their tests make sense. I also ran the regression on all the repos and confirmed there were no differences and all the logs looked the same for the few that had errors. So I'm approving!

**Review by ngfreiter — APPROVED** [review]
Looks good to me!

**Comment by cmoesel** [comment]
NOTE: Based on a new understanding of the behavior of content references, much of this PR was _backed out_ in #753.

**Pull request merged**
